src/txmempool.cpp

```diff
@@ -331,7 +331,7 @@ void CTxMemPoolEntry::UpdateAncestorState(int64_t modifySize, CAmount modifyFee,
     assert(int(nSigOpCostWithAncestors) >= 0);
 }
 
-CTxMemPool::CTxMemPool(CBlockPolicyEstimator* estimator)
+CTxMemPool::CTxMemPool(std::shared_ptr<CBlockPolicyEstimator> estimator)
```

Comment:
You don't need to make this change. The fee estimator's memory/lifetime is owned by `NodeContext`. This can continue being a raw pointer to that object (in the same way that `PeerManager` holds pointers and references to other components).

Not making this change should reduce the diff of this PR considerably.